var/spack/repos/builtin/packages/gmsh/package.py

```diff
@@ -144,3 +139,27 @@ def cmake_args(self):
             options.append('-DENABLE_COMPRESSED_IO:BOOL=ON')
 
         return options
+    
+    def setup_build_environment(self, env):
+        if "+oce" in self.spec:
+            env.prepend_path('CASROOT', self.spec['oce'].prefix)
+        if "+petsc" in self.spec:
+            env.prepend_path('PETSC_DIR', self.spec['petsc'].prefix)
+        if "+tetgen" in self.spec:
+            env.prepend_path('TETGEN_DIR', self.spec['tetgen'].prefix)
+        if "+netgen" in self.spec:
+            env.prepend_path('NETGEN_DIR', self.spec['netgen'].prefix)
+        if "+slepc" in self.spec:
+            env.prepend_path('SLEPC_DIR', self.spec['slepc'].prefix)
+
+    def setup_run_environment(self, env):
```

Comment:
Seems like this is only needed for CMake, so we would actually not need the variables in the runtime environment?

Reply:
so setup_run_environment should be removed? I am not sure I have the full picture here

Reply:
I'd think so. Most of this stuff looks like it is for CMake, not actually needed when executing anything built with GMSH